### DISCUSSION TITLE
Client stays connected even with a session timeout?

### DISCUSSION THREAD

**Discussion by Mungoid** [FreeRDP/FreeRDP#8228] [Q&A]
I'm not very knowledgeable about RDP in general so apologies if this is basic, I'm just curious on the differences I'm seeing in session timeouts. 

On Windows when I connect to my work using MSTSC, it consistently disconnects users after ~9 hours of connection, or ~30 min of inactivity. However, on Linux Mint 20.x using FreeRDP (forgot which version), I always got disconnected after about 5 min of inactivity, period. I assume its when my remote work PC screen locks but not sure. 

Recently I updated to Linux Mint 21, and now FreeRDP(2.6.1) seems to never disconnect even on my remote PC lock screen. I started my session yesterday morning, and noticed this morning that it is still connected. And I'm still using the exact same command as I did for LM 20.x:

`xfreerdp +nego /gt:http /dynamic-resolution /v:xxx.xxx.xx.xxx /u:me@blah.com /cert:ignore /log-level:info /g:remote.blah.com`

I always assumed the timeout was controlled by my works RDP gateway, but now I'm wondering if it is the client end. Or maybe Group Policy. My home Windows install isn't in my works domain, so I'm not sure. Or maybe my work updated their gateway to handle things differently? 

Thoughts?

**Comment by akallabeth** [comment]
hard to tell without knowing the server configuration.
As you assumed correctly, these timeouts are enforced by the server (or gateway)
but they depend on configurable parameters.
Some of these parameters can be last client interaction, others are tailored to specific clients (identification is not always correct)
